xarray/tests/test_backends.py

```diff
@@ -609,8 +609,8 @@ def test_orthogonal_indexing(self):
             assert_identical(expected, actual)
 
     @pytest.mark.xfail(
-        not has_dask,
-        reason="the code for indexing without dask handles negative steps in slices incorrectly",
+        has_dask or not has_dask,
+        reason="the code for indexing with or without dask handles negative steps in slices incorrectly",
```

Comment:
`open_zarr` uses `chunks="auto"` by default, whereas `open_dataset` defaults to `chunks=None` (they have quite different behaviour in the Zarr backend). I think this is a problem because using `chunks=None` with `engine="zarr"` will automatically convert the arrays to numpy arrays upon access (see http://xarray.pydata.org/en/v0.15.1/generated/xarray.open_zarr.html ), instead of keeping them as dask arrays, and that might explain why this trivial test fails when refactoring from `open_zarr` to `open_dataset(engine="zarr")`.

Reply:
I've updated the Zarr tests to use `chunks="auto"` as was the default in `open_zarr`, and so `test_vectorized_indexing` works properly for negative slices when `dask` is installed (see commit dce4e7cd1fcf35fb7d3293bf6cc410646b588c64 and 31ce87d75834ddab6466a013c1aae4d55a450b3a). Happy to resolve this, but there should be a discussion around whether `open_mfdataset` with `engine="zarr"` should use `chunks="auto"` as in `open_zarr`, or `chunks=None` as with the other backends (e.g. netcdf, h5netcdf, etc).